go/vt/vtgate/vtgate_test.go

```diff
@@ -509,3 +509,117 @@ func TestErrorIssuesRollback(t *testing.T) {
 	}
 	sbc.MustFailCodes[vtrpcpb.Code_ALREADY_EXISTS] = 0
 }
+
+var shardedVSchema = `
+{
+	"sharded": true,
+	"vindexes": {
+		"hash_index": {
+			"type": "hash"
+		}
+	},
+	"tables": {
+		"user_extra": {
+			"column_vindexes": [
+				{
+					"column": "user_id",
+					"name": "hash_index"
+				}
+			]
+		}
+	}
+}
+`
+
+func TestMultiInternalSavepointVtGate(t *testing.T) {
+	s := createSandbox(KsTestSharded)
+	s.ShardSpec = "-40-80-"
+	s.VSchema = shardedVSchema
+	srvSchema := getSandboxSrvVSchema()
```

Comment:
I think we need to change this so we can use `shardedVSchema` instead of the default sandbox's vschema.

Reply:
The vschema always needs to be assigned (otherwise it will remain empty), there is a default shardSpec that needs to be overridden.